edk2toolext/windows/secureboot/secureboot_audit.py

```diff
@@ -0,0 +1,579 @@
+# @file This file tests secureboot_audit.py script
+# Command-line tool for inspecting UEFI Secure Boot databases
+# Requires "pip install edk2-pytool-extensions"
+##
+# Copyright (c) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+"""Command-line tool for inspecting UEFI Secure Boot databases."""
+
+import argparse
+import logging
+import sys
+import ctypes
+import os
+import json
+import xlsxwriter
+import openpyxl
+
+# Some of these libraries are windows only, so we need to import them conditionally
+if sys.platform == "win32":
+    import win32api
+    import win32process
+    import win32security
+    import pywintypes
+
+# import this from edk2toollib so we can parse the dbx
+from edk2toollib.uefi.authenticated_variables_structure_support import (
+    EfiSignatureDatabase,
+)
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+if sys.platform == "win32":
+    KERNEL32 = ctypes.windll.kernel32
+    EFI_VAR_MAX_BUFFER_SIZE = 1024 * 1024
+
+DEFAULT_OUTPUT_FOLDER = os.path.join(".", "SecureBootFiles")
+
+# The supported files to retrieve from UEFI and their guids, additional files can be added here
+SECUREBOOT_FILES = {
+    "dbx": "{d719b2cb-3d3a-4596-a3bc-dad00e67656f}"  # EFI_IMAGE_SECURITY_DATABASE_GUID
+}
+
+KNOWN_CERTIFICATES = [
+    "11FB1A56B8CAE09CF7B06AD620A9F13C604D108BDAF443C115FE5848805241E5",  # "Canonical Certificate",
+    "F22246C9E65BB4543CB892B6507C062196A788E5A190D2D5A877B94DFDFB2935",  # "Debian Certificate",
+]
+
+
+###################################################################################################
+# Helper functions
+###################################################################################################
+
+
+def write_json_file(report, output_file):
```

Comment:
Sorry doug, I missed this the first time. Please also add type hinting to the function arguments :(